Source/DafnyDriver/DafnyDoc.cs

```diff
@@ -0,0 +1,1240 @@
+    nameIndex.Add(name + " " + nameIndex.Count + " " + owner, kind + " " + QualifiedNameWithLinks(target));
+  }
+
+  public void WriteIndex() {
+    var keys = nameIndex.Keys.ToList();
+    keys.Sort();
+
+    string filename = Outputdir + "/nameindex.html";
+    using StreamWriter file = new(filename);
+    file.Write(head1);
+    file.Write($"Index for program {DafnyProgram.Name}");
+    file.Write(head2);
+    file.Write(style);
+    file.Write($"<div>\n<h1>Index{ProgramHeader()}{space4}{Smaller(contentslink)}</h1>\n</div>");
+    file.Write(bodystart);
+    foreach (var key in keys) {
+      var k = key.IndexOf(' ');
+      var value = nameIndex[key]; // Already rewritten as a bunch of links
+      var keyn = key.Substring(0, k);
+      k = key.LastIndexOf(' ');
+      var owner = key.Substring(k + 1);
+      var hash = value.IndexOf('#');
+      if (hash == -1) {
+        file.Write($"<a href=\"{owner}.html\">{keyn}</a>");
+      } else if (value.StartsWith("export")) {
+        file.Write($"<a href=\"{owner}.html#{ExportSetAnchor(keyn)}\">{keyn}</a>");
+      } else {
+        var link = value.Substring(0, hash);  // TODO - what is link for // TODO _ do export sets work
+        file.Write($"<a href=\"{owner}.html#{keyn}\">{keyn}</a>");
+      }
+      file.WriteLine(Mdash + value + br);
+    }
+    file.Write(foot);
+    AnnounceFile(filename);
+  }
+
+  public static string DefaultOutputDir = "./docs";
+
+  static string eol = "\n";
+
+  static string br = "<br>";
+
+  public static string Mdash = " &mdash; ";
+
+  static string contentslink = "<a href=\"index.html\">[table of contents]</a>";
+  static string indexlink = "<a href=\"nameindex.html\">[index]</a>";
+
+
+  static string space4 = "&nbsp;&nbsp;&nbsp;&nbsp;";
+
+  static string indent = "<p style=\"margin-left: 25px;\">";
+  static string unindent = "</p>";
+  static string head1 =
+  @"<!doctype html>
+
+<html lang=""en"">
+<head>
+  <meta charset=""utf-8"">
+  <meta name=""viewport"" content=""width=device-width, initial-scale=1"">
+
+  <title>";
+
+  static string head2 =
+  @"</title>
+  <link rel=""icon"" type=""image/png"" href=""dafny-favicon.png"">
+  <meta name=""description"" content=""Documentation for Dafny code produced by dafnydoc"">
+  <meta name=""author"" content=""dafnydoc"">
+</head>
+";
+
+  static string style =
+  @"<style>
+body {
+  background-color: white;
+}
+
+h1 {
+  color: blue;
+  text-align: center;
+  background-color: #fceb6c
+}
+h2 {
+  color: blue;
+  text-align: left;
+  background-color: #fceb6c
+}
+h3 {
+  color: blue;
+  text-align: left;
+  background-color: #fefdcc
+}
+
+p {
+  font-size: 16px
+}
+</style>
+";
+
+  static string bodystart =
+  @"<body>
+";
+
+  public static string foot =
+  @"</body>
+</html>
+";
+
+}
+
+public static partial class Extensions {
+  /// <summary>
+  ///     A string extension method that replace first occurence.
+  /// </summary>
+  /// <param name="this">The @this to act on.</param>
```

Comment:
The `<param>` documentation doesn't seem to say more than the names of these parameters, so it seems better to leave it out.